File: src/class-problems/count-the-number-of-consistent-strings/1684.js
// 1684. Count the Number of Consistent Strings

/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
var countConsistentStrings = function (allowed, words) {
  let counter = words.length;
  let allowedSet = new Set(allowed);

  for (word of words) {
    for (char of word) {
      if (!allowedSet.has(char)) {
        counter--;
        break;
      }
    }
  }
  return counter;
};

File: src/leetcode/count-the-number-of-consistent-strings/1684.py
#1684. Count the Number of Consistent Strings

class Solution(object):
    def countConsistentStrings(self, allowed, words):
        """
        :type allowed: str
        :type words: List[str]
        :rtype: int
        """

        allowed = set(allowed)
        result = 0

        for word in words:
            if all(char in allowed for char in word):
                result += 1

        return result